newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectRecoveryDialog.js

```diff
@@ -0,0 +1,267 @@
+      )
+    : null;
+  const cloudProjectName = cloudProject ? cloudProject.name : null;
+
+  return (
+    <I18n>
+      {({ i18n }) => (
+        <Dialog
+          open
+          flexColumnBody
+          maxWidth="sm"
+          onRequestClose={onClose}
+          onApply={onApply}
+          actions={actions}
+          title={
+            lastSaneVersion && !isLoading ? (
+              <Trans>A functioning save has been found!</Trans>
+            ) : (
+              <Trans>An issue was found on this project</Trans>
+            )
+          }
+        >
+          {isErrored ? (
+            <AlertMessage kind="error">
+              <Trans>
+                An error occurred while trying to recover your project last
+                versions. Please try again later.
+              </Trans>
+            </AlertMessage>
+          ) : isLoading ? (
+            <>
+              <Line>
+                <Column noMargin>
+                  <Text noMargin>
+                    <Trans>
+                      The latest save of{' '}
+                      {cloudProjectName ? (
+                        `"${cloudProjectName}"`
+                      ) : (
+                        <Trans>this project</Trans>
+                      )}{' '}
+                      is corrupted and cannot be opened.
+                    </Trans>
+                  </Text>
+                  <Text noMargin>
+                    <Trans>
+                      Please wait while we scan your project to find a solution.
+                    </Trans>
+                  </Text>
+                </Column>
+              </Line>
+              <Line justifyContent="center">
+                <CircularProgress />
+              </Line>
+            </>
+          ) : saneVersionHasNotBeenFound ? (
+            <>
+              <Line>
+                <Column noMargin>
+                  <Text noMargin>
+                    <Trans>We couldn't find a version to go back to.</Trans>
+                  </Text>
+                  <Text noMargin>
+                    <Trans>
+                      Please get in touch with us to find a solution.{' '}
+                    </Trans>
+                  </Text>
+                </Column>
+              </Line>
+            </>
+          ) : lastSaneVersion ? (
+            <>
+              <Line>
+                <Column noMargin>
+                  <Text noMargin>
+                    <Trans>
+                      The latest save of{' '}
+                      {cloudProjectName ? (
+                        `"${cloudProjectName}"`
+                      ) : (
+                        <Trans>this project</Trans>
+                      )}{' '}
+                      is corrupted and cannot be opened.
+                    </Trans>
+                  </Text>
+                  <Text noMargin>
+                    <Trans>
+                      However, we have found a non-corrupted save from{' '}
```

Comment:
```suggestion
                      We have found a non-corrupted save from{' '}
```

Too much verbose. 
And not in the mockup.